sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -846,14 +845,30 @@ def upload_blob(self, repository, data, **kwargs):
             if repository is None or data is None:
                 raise ValueError("The parameter repository and data cannot be None.")
             raise
-        return complete_upload_response_headers['Docker-Content-Digest']
+        return complete_upload_response_headers['Docker-Content-Digest'], blob_size
+
+    def _upload_blob_chunk(self, location: str, data: IO, **kwargs) -> Tuple[str, str, int]:
+        hasher = hashlib.sha256()
+        buffer = data.read(DEFAULT_CHUNK_SIZE)
+        blob_size = len(buffer)
+        while len(buffer) > 0:
+            response_headers = cast(Dict[str, str], self._client.container_registry_blob.upload_chunk(
+                location,
+                buffer,
+                cls=_return_response_headers,
+                **kwargs
+            ))
+            location = response_headers['Location']
+            hasher.update(buffer)
+            buffer = data.read(DEFAULT_CHUNK_SIZE)
+            blob_size += len(buffer)
+        return "sha256:" + hasher.hexdigest(), location, blob_size
 
     @distributed_trace
-    def download_manifest(self, repository, tag_or_digest, **kwargs):
-        # type: (str, str, **Any) -> DownloadManifestResult
+    def download_manifest(self, repository: str, tag_or_digest: str, **kwargs) -> DownloadManifestResult:
         """Download the manifest for an OCI artifact.
 
-        :param str repository: Name of the repository
+        :param str repository: Name of the repository.
         :param str tag_or_digest: The tag or digest of the manifest to download.
         :returns: DownloadManifestResult
         :rtype: ~azure.containerregistry.models.DownloadManifestResult
```

Comment:
It looks like you are fixing the Accept header to "OCI_MANIFEST_MEDIA_TYPE" below - I believe that as a feature crew we would prefer to request all known manifest media types to ensure that we get the manifest back in the expected format, without the service converting it to a different format.  Let me know if you have any questions on this!

Reply:
Thanks for your review @annelo-msft ! You're right, I'm preparing another PR for the manifest custom media type support, I'll tag you once the PR is ready. This is one just focusing on blob.